changelogs/unreleased/gh-7473-introduce-system-fiber.md

```diff
@@ -0,0 +1,5 @@
+## feature/core
+
+* Introduced system fibers, which cannot be cancelled from Lua. Added C API
+  for setting fiber as system: `fiber_set_system()`, in order to check if
+  the fiber is not cancellable `fiber:is_system()` can be used now (gh-7473)
```

Comment:
I wouldn't expose this to public API. The way to make a fiber non-system makes this patch nop, because I can just call `fiber_set_system(false)` on any "system" fiber and then cancel it. And it will work just like without this patchset at all - unexpected problems with internal cancelled fibers.